src/main/java/org/pih/petl/api/EtlService.java

```diff
@@ -97,6 +103,16 @@ public JobExecution getLatestJobExecution(String jobPath) {
         return l.get(0);
     }
 
+    public PetlJob getPetlJob(JobConfig jobConfig) {
+        for (PetlJob petlJob : petlJobs) {
+            Component component = petlJob.getClass().getAnnotation(Component.class);
+            if (component != null && jobConfig.getType().equals(component.value())) {
+                return petlJob;
+            }
+        }
+        throw new PetlException("Unknown job type: " + jobConfig.getType());
+    }
+
```

Comment:
I got rid of the PetlJobFactory in favor of making all of the different Jobs into components.